### PR TITLE
add a readme for the dart fix tests

### PR DESCRIPTION
**Pull request by devoncarew**
- add a readme for the dart fix tests written for package:flutter

This PR adds a readme for the dart fix tests (https://github.com/flutter/flutter/tree/master/packages/flutter/test_fixes), and calls out that there is an upstream dependency on the tests.

This PR does not effect any issues, and, as a readme only change I suspect it's exempt from the test requirement.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test_fixes/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+data driven fixes.
+
+## When making structural changes to this directory
+
+Note that the tests in this directory are also invoked from external repositories.
+Specifically, the CI system for the dart-lang/sdk repo runs these tests in order to
+ensure that changes to the dart fix file format do not break Flutter.
+
+See [tools/bots/flutter/analyze_flutter_flutter.sh](https://github.com/dart-lang/sdk/blob/master/tools/bots/flutter/analyze_flutter_flutter.sh)
+for the where the tests are invoked.
```
Comment (MrBirb) [inline packages/flutter/test_fixes/README.md:18]: nit: redundant the*

Comment (devoncarew) [reply packages/flutter/test_fixes/README.md]: Thanks! Updated.

**Review by goderbauer — APPROVED** [review]
LGTM

**Review by Piinks — APPROVED** [review]
LGTM! Thanks for adding this! :)

**Pull request merged**